src/parser/parser.c

```diff
@@ -37,7 +39,7 @@ t_cmd_info	*parser(t_parser_info p_info, t_list **malloc_list)
 	}
 	cmd->type = BIN;
 	err = set_io_files(p_info.line, cmd, malloc_list);
-	if (err == -1)
+	if (err != 0)
```

Comment:
I guess in this case we still need a possibility to pass that error up or set it as an key in our cmd or app struct